File: Sdk-ObjectiveC/office365-base-sdk/office365-base-sdk/FileDiscoveryClient.m
//
//  FileDiscoveryClient.m
//  office365-base-sdk
//
//  Created by Gustavo on 7/21/14.
//  Copyright (c) 2014 Lagash. All rights reserved.
//

#import "FileDiscoveryClient.h"
#import "DiscoveryInformation.h"
#import "HttpConnection.h"
#import "Constants.h"

@implementation FileDiscoveryClient


- (instancetype)initWithUrl:(NSString *)url credentials:(Credentials *)credentials{
    self =[super initWithUrl:url credentials:credentials];
    return self;
}

- (NSURLSessionDataTask *)getDiscoveryInfo:(void (^)(NSData *data, NSURLResponse *response, NSError *error))callback
{
    HttpConnection *connection = [[HttpConnection alloc] initWithCredentials:self.Credential
                                                                       url: @"https://api.office.com/discovery/me/services"];

    NSString *method = (NSString *)[[Constants alloc] init].Method_Get;
    return [connection execute:method callback:callback];
}

- (NSMutableArray *)parseData:(NSData *)data{
    NSMutableArray *array = [NSMutableArray array];
    //NSError *err;
    
    NSDictionary *jsonResult = [NSJSONSerialization JSONObjectWithData:data
                                                               options: NSJSONReadingMutableContainers
                                                                 error:nil];
    
    NSArray *jsonArray  = [[jsonResult valueForKey:@"d"] valueForKey:@"results"];
    
    for (NSDictionary *value in jsonArray) {
        DiscoveryInformation *information = [[DiscoveryInformation alloc] init];
        [information createFromJson:value];
        
        [array addObject:information];
    }
    
    return array;
}

@end

File: Sdk/office365_base_sdk/office365_base_sdk/FileDiscoveryClient.swift
//
//  FileDiscoveryClient.swift
//  office365-base-sdk
//
//  Created by Lagash on 6/13/14.
//  Copyright (c) 2014 microsoft. All rights reserved.
//

import Foundation

class FileDiscoveryClient: BaseClient{
    
    func getDiscoveryInfo(callback : ((NSData!, NSURLResponse!, NSError!) -> Void)!) -> NSURLSessionDataTask {
     
        var connection = HttpConnection(credentials: self.Credential, url: "https://api.office.com/discovery/me/services");
        
        return connection.execute(Constants.Method_Get, callback : callback);
    }
      
    override func parseData(data : NSData) -> NSMutableArray{
        var array = NSMutableArray();
        var err: NSError;
        
        var jsonResult: NSDictionary = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.MutableContainers, error: nil) as NSDictionary;
        
        var jsonArray : NSArray = (jsonResult.valueForKey("d") as NSDictionary).valueForKey("results") as NSArray;
        
        for value : AnyObject in jsonArray{
            var dinformation = DiscoveryInformation();
            
            dinformation.createFromJson(value as NSDictionary)

            array.addObject(dinformation);
        }
        
        return array;
    }
}